optimizely/event_builder.py

```diff
@@ -139,54 +145,57 @@ class EventParams(object):
     CUSTOM = 'custom'
     ANONYMIZE_IP = 'anonymize_ip'
 
-  def _add_attributes(self, attributes):
-    """ Add attribute(s) information to the event.
+  def _get_attributes(self, attributes):
+    """ Get attribute(s) information.
 
     Args:
       attributes: Dict representing user attributes and values which need to be recorded.
     """
 
-    visitor = self.params[self.EventParams.USERS][0]
-    visitor[self.EventParams.ATTRIBUTES] = []
+    params = []
 
     if not attributes:
-      return
+      return []
 
     for attribute_key in attributes.keys():
       attribute_value = attributes.get(attribute_key)
-      # Omit falsy attribute values
-      if attribute_value:
+      # Do not discard if value is zero or false
+      if attribute_value is not None:
```

Comment:
attribute value can be integer zero or boolean false.

Reply:
Attributes are supposed to be strings always. So the previous check was valid.

Reply:
according to documentation, it can accept string, boolean and int. right now, no test case when attributes value should be  like {"is_local": false} which would be discarded by old code but actually it shouldn't be discarded.

Reply:
@msohailhussain it should be discarded and the existing behavior is correct.

Reply:
ok sure, we will revert it.